src/jdk.incubator.concurrent/share/classes/jdk/incubator/concurrent/ScopedValue.java

```diff
@@ -0,0 +1,839 @@
+                ScopedValue.Cache.invalidate(bitmask);
+            }
+        }
+
+        /**
+         * Runs an operation with each scoped value in this mapping bound to its value
+         * in the current thread.
+         * When the operation completes (normally or with an exception), each scoped value
+         * in the mapping will revert to being unbound, or rervert to its previous value
+         * when previously bound, in the current thread.
+         *
+         * <p> Scoped values are intended to be used in a <em>structured manner</em>.
+         * If {@code op} creates a {@link StructuredTaskScope} but does not {@linkplain
+         * StructuredTaskScope#close() close} it, then exiting {@code op} causes the
+         * underlying construct of each {@code StructuredTaskScope} created in the
+         * dynamic scope to be closed. This may require blocking until all child threads
+         * have completed their sub-tasks. The closing is done in the reverse order that
+         * they were created. Once closed, {@link StructureViolationException} is thrown.
+         *
+         * @param op the operation to run
+         * @see ScopedValue#where(ScopedValue, Object, Runnable)
+         */
+        public void run(Runnable op) {
+            Objects.requireNonNull(op);
+            Cache.invalidate(bitmask);
+            var prevSnapshot = scopedValueBindings();
+            var newSnapshot = new Snapshot(this, prevSnapshot);
+            runWith(newSnapshot, op);
+        }
+
+        /**
+         * Execute the action with a set of ScopedValue bindings.
+         *
+         * The VM recognizes this method as special, so any changes to the
+         * name or signature require corresponding changes in
+         * JVM_FindScopedValueBindings().
+         */
+        @Hidden
+        @ForceInline
+        private void runWith(Snapshot newSnapshot, Runnable op) {
+            try {
+                JLA.setScopedValueBindings(newSnapshot);
+                JLA.ensureMaterializedForStackWalk(newSnapshot);
+                ScopedValueContainer.run(op);
+            } finally {
+                Reference.reachabilityFence(newSnapshot);
+                JLA.setScopedValueBindings(newSnapshot.prev);
+                Cache.invalidate(bitmask);
+            }
+        }
+    }
+
+    /**
+     * Creates a new {@code Carrier} with a single mapping of a {@code ScopedValue}
+     * <em>key</em> to a value. The {@code Carrier} can be used to accumlate mappings so
+     * that an operation can be executed with all scoped values in the mapping bound to
+     * values. The following example runs an operation with k1 bound (or rebound) to v1,
+     * and k2 bound (or rebound) to v2.
+     * {@snippet lang=java :
+     *     // @link substring="run" target="Carrier#run(Runnable)" :
+     *     ScopedValue.where(k1, v1).where(k2, v2).run(() -> ... );
+     * }
+     *
+     * @param key the ScopedValue key
```

Comment:
should use `@code` or `@link`

Reply:
Sorry, I don't understand what you want.

Reply:
There is a "bare" reference to `ScopedValue`. Should it be `{@code ScopedValue}` or `{@link ScopedValue}` ?

Reply:
Ahh, OK.

Reply:
We fixed those in the recent update.